src/debug_clang.h

```diff
@@ -0,0 +1,85 @@
+#ifndef JUCI_DEBUG_CLANG_H_
+#define JUCI_DEBUG_CLANG_H_
+
+#include <boost/filesystem.hpp>
+#include <unordered_map>
+#include <lldb/API/SBDebugger.h>
+#include <lldb/API/SBListener.h>
+#include <lldb/API/SBProcess.h>
+#include <thread>
+#include <mutex>
+
+namespace Debug {
+  class Clang {
```

Comment:
I'd like to see more abstraction here. The methods defined in this class are useful in every debugger implementation (and which are called by the UI callbacks). If we were to expand to other debug implementations it's better to abstract now I guess.